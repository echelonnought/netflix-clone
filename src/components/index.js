export { default as Header } from './header/Index';
export { default as Feature } from './feature/Index';
export { default as Accordion } from './accordion/Index';
export { default as OptForm } from './opt-form/Index';
export { default as Jumbotron } from './jumbotron/Index';
export { default as Footer } from './footer/Index';
export { default as SigninForm } from './signin-form/Index';
export { default as Profiles } from './profiles/Index';
export { default as Loading } from './loading/Index';
export { default as Card } from './card/Index';
export { default as Player } from './player/Index';