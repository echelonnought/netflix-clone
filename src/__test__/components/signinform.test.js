import React from 'react';
import { render } from '@testing-library/react';
import { SigninForm } from '../../components';

jest.mock('react-router-dom');

describe('<SigninForm />', () => {
    it('renders the <SigninForm /> with populated data', () => {
       const { container, getByText, getByPlaceholderText } = render (
        <SigninForm >
        <SigninForm.Title>Sign In Now</SigninForm.Title>
        <SigninForm.Base >
              <SigninForm.Input 
              type="email" 
              placeholder="Email address" 
              onChange={() => {}}/>

              <SigninForm.Input 
              type="password" 
              placeholder="Password" 
              required 
              onChange={() => {}}/>
              
              <SigninForm.Submit disabled type="submit">
                  Sign In
              </SigninForm.Submit>
        </SigninForm.Base>
        <SigninForm.Text>
            New to Netflix? <SigninForm.Link to="/signup">Sign up now</SigninForm.Link>
        </SigninForm.Text>
        <SigninForm.TextSmall>This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more</SigninForm.TextSmall>
      </SigninForm >
       )
    expect(getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more")).toBeTruthy();
    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <SigninForm /> with an error', () => {
        const { container, getByText, queryByText } = render(
            <SigninForm>
                <SigninForm.Error>Your email address is already being used</SigninForm.Error>
                <SigninForm.Submit type="submit">Sign In</SigninForm.Submit>
            </SigninForm>
        )
        expect(getByText('Your email address is already being used')).toBeTruthy();
        expect(queryByText('Sign In').disabled).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    })
    
});
