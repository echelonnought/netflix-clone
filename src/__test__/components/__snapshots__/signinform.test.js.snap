// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigninForm /> renders the <SigninForm /> with an error 1`] = `
<div
  class="signin-form__Container-sc-1wz9jml-0 biiEYX"
>
  <div
    class="signin-form__Error-sc-1wz9jml-1 gzSnfZ"
  >
    Your email address is already being used
  </div>
  <button
    class="signin-form__Submit-sc-1wz9jml-8 flXSCG"
    type="submit"
  >
    Sign In
  </button>
</div>
`;
